### PR TITLE
Fix lint warnings/no-warning-comments

### LINKED ISSUE #608: Fix lint warnings
The following rules were set to "warn" so that we could focus on more important issues, but these should all be fixed as well:

- [ ] no-negated-condition
- [ ] no-unused-vars
- [ ] prefer-destructuring
- [ ] unicorn/no-for-loop
- [ ] unicorn/prefer-includes
- [ ] unicorn/prefer-string-slice
- [ ] unicorn/filename-case
- [ ] unicorn/prefer-query-selector

The overrides are defined in https://github.com/zaproxy/zap-hud/blob/develop/package.json#L54-L63 (the lines may change in time)

### PR DESCRIPTION
**Pull request by njmulsqb**
Fixed couple of warnings.

Fixes partially #608

### CODE REVIEW

**Comment by psiinon** [comment]
In general we shouldnt just delete TODOs without actually adressing the things they were about .. but in this case I think its reasonable :)

**Review by psiinon — APPROVED** [review]

**Comment by thc202** [comment]
From which linter is that?

**Comment by njmulsqb** [comment]
> In general we shouldnt just delete TODOs without actually adressing the things they were about .. but in this case I think its reasonable :)

Actually the linter says that such comments should not exist in a production software, what should I do with rest of the todo comments that are causing warnings?

**Comment by njmulsqb** [comment]
> From which linter is that?

Default linter of hud, I guess eslint. Got the warnings by _./gradlew npmLintAllHud_

**Comment by thc202** [comment]
Thank you. Yes, ideally we should fix them all, either address the todo or remove if no longer relevant.

File: src/main/zapHomeFiles/hud/tools/history.js [hunk 1]
```diff
@@ -6,14 +6,13 @@
 
 const History = (function () {
 	// Constants
-	// todo: could probably switch this to a config file?
```
Comment (thc202) [inline src/main/zapHomeFiles/hud/tools/history.js:9]: This and the following can be removed from all tools.

**Comment by njmulsqb** [comment]
> Thank you. Yes, ideally we should fix them all, either address the todo or remove if no longer relevant.

How can I be sure that a particular todo is no longer relevant?

File: src/main/zapHomeFiles/hud/tools/history.js (hunk 1 above)
Comment (njmulsqb) [reply src/main/zapHomeFiles/hud/tools/history.js:9]: Ok I will

**Comment by thc202** [comment]
Thought question, it really depends of what the todo is about :) If not sure just ask.

**Comment by njmulsqb** [comment]
> Working to remove ` // Todo: change this to a util function that reads in a config file (json/xml)` now from code

Removed all of its occurrences

**Comment by thc202** [comment]
Thank you, could you revert the changes done to the lock file and fix up all the commits?

**Comment by njmulsqb** [comment]
I have reverted changes in package-lock.json file and fixed up last 2 commits.
Should package-lock.json be added to .gitignore to avoid pushing changes again? Running `npm i` updates this file

**Comment by njmulsqb** [comment]
These are the pending no-warning-comments in HUD.

    // TODO change to try loading from localstorage (src\main\zapHomeFiles\hud\tools\commonAlerts.js)
    // Todo: change this to 'continue' and figure out / fix stopBreaking (src\main\zapHomeFiles\hud\tools\break.js)
    // FIXME: remove after  https://github.com/zaproxy/zap-hud/pull/620 (src\main\zapHomeFiles\hud\display.js)
    // Todo: is this too hacky? (src\main\zapHomeFiles\hud\target\inject.js)
    // TODO showComments section - put this code in a separate file and inject ? (src\main\zapHomeFiles\hud\target\inject.js)
    // TODO showEnable section - put this code in a separate file and inject ? (src\main\zapHomeFiles\hud\target\inject.js)

Give me the go-ahead to remove them and then we can merge this PR

**Comment by thc202** [comment]
Better leave those for now. The commits were not fixed up yet.

**Comment by njmulsqb** [comment]
> Better leave those for now. The commits were not fixed up yet.

I have merged the 2 consequent commits, but the earlier ones are from Oct last year, after which many more commits have been made as visible via `git log --oneline` how can I merge non-consecutive commits?

**Comment by thc202** [comment]
The following should do it:
```
git fetch upstream
git rebase -i upstream/main
```
Set the last two commits to fixup and done.

**Comment by njmulsqb** [comment]
> git rebase -i upstream/main

I have these commits,
pick 91fe5c6 no-warning-comments resolved
pick 62222b2 Some more warning comments removed
pick 319133f build(deps): bump vue-i18n from 8.26.5 to 8.26.7
pick 90427c5 build(deps-dev): bump eslint-plugin-no-unsanitized from 3.2.0 to 4.0.1
pick c5794ea fix: remove some TODO comments to solve lint warnings
The last two were already fixed up. Are you talking about first two?

**Comment by thc202** [comment]
Remove the lines containing `319133f` and `90427c5` and set `62222b2` and `c5794ea` to `fixup`.

**Comment by njmulsqb** [comment]
I think we're done with all the no-warning-comments now except the 4 as mentioned previously.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**